test/grpc-fallback.ts

```diff
@@ -0,0 +1,216 @@
+
+    assert(echoStub instanceof protobuf.rpc.Service);
+
+    // The stub should consist of service methods
+    expect(echoStub.echo).to.be.a('Function');
+    expect(echoStub.pagedExpand).to.be.a('Function');
+    expect(echoStub.wait).to.be.a('Function');
+
+    // There should be 6 methods for the echo service (and 4 other methods in the object)
+    assert.strictEqual(Object.keys(echoStub).length, 10);
+
+    // Each of the service methods should take 4 arguments (so that it works with createApiCall)
+    assert.strictEqual(echoStub.echo.length, 4);
+  });
+
+  it('should support optional parameters', async () => {
+    // tslint:disable-next-line no-any
+    const echoStub: any = await gaxGrpc.createStub(
+      echoService,
+      stubExtraOptions
+    );
+
+    assert(echoStub instanceof protobuf.rpc.Service);
+
+    // The stub should consist of methods
+    expect(echoStub.echo).to.be.a('Function');
+    expect(echoStub.collect).to.be.a('Function');
+    expect(echoStub.chat).to.be.a('Function');
+
+    // There should be 6 methods for the echo service (and 4 other members in the object)
+    assert.strictEqual(Object.keys(echoStub).length, 10);
+
+    // Each of the service methods should take 4 arguments (so that it works with createApiCall)
+    assert.strictEqual(echoStub.echo.length, 4);
+  });
+});
+
+describe('grpc-fallback', () => {
+  let gaxGrpc, protos, echoService, stubOptions;
+  const createdAbortControllers = [];
+  // @ts-ignore
+  const savedAbortController = abortController.AbortController;
+
+  before(() => {
+    stubOptions = {
+      servicePath: 'foo.example.com',
+      port: 443,
+    };
+
+    // @ts-ignore incomplete options
+    gaxGrpc = new GrpcClient(opts);
+    protos = gaxGrpc.loadProto(echoProtoJson);
+    echoService = protos.lookupService('Echo');
+    stubOptions = {
+      servicePath: 'foo.example.com',
+      port: 443,
+    };
+
+    //tslint:disable-next-line variable-name
+    const AbortController = function() {
+      // @ts-ignore
+      this.abort = function() {
+        // @ts-ignore
+        this.abortCalled = true;
+      };
+      // @ts-ignore
+      createdAbortControllers.push(this);
+    };
+
+    // @ts-ignore
+    abortController.AbortController = AbortController;
+  });
+
+  beforeEach(() => {
+    createdAbortControllers.splice(0);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  after(() => {
+    // @ts-ignore
+    abortController.AbortController = savedAbortController;
+  });
+
+  it('should make a request', async () => {
+    const requestObject = {content: 'test-content'};
+    const responseType = protos.lookupType('EchoResponse');
+    const response = responseType.create(requestObject); // request === response for EchoService
+
+    sinon.stub(nodeFetch, 'Promise').returns(
+      Promise.resolve({
+        arrayBuffer: () => {
+          Promise.resolve(responseType.encode(response).finish());
```

Comment:
I think this should be `return Promise.resolve(responseType.encode(response).finish());` just like we discovered in the browser tests.

Reply:
Yes, already fixed that right after that discussion, thank you!